### PR TITLE
Add SpaceNet6 datamodule

### PR DESCRIPTION
**Pull request by nilsleh**
This PR adds a DataModule for SpaceNet6. And makes the following changes:
- Introduce `SpaceNetBaseDataModule`
- Add dummy test data for `SpaceNet1`, it exists in the actual dataset, but not in the dummy data
- Add dummy data for `SpaceNet6`
- Resctructure `tests/data/spacenet/` to generate test data for each version

Other Notes:
- The "test" data in `SpaceNet` comes without labels, so in the datamodules, I have configured them to be "predict" data.
- One could also restructure the `data.py` scripts, because there are shared functions that can be utilized across versions.
- it would be nice to have geospatial train/val/test splits generated from the `train` data, and not random splits

### CODE REVIEW

File: tests/datasets/test_spacenet.py
Comment (adamjstewart) [inline tests/datasets/test_spacenet.py:1]: Now that we have fake data for SpaceNet 6, should we also test that here? Should be able to parametrize the fixture such that you don't need to add any extra lines of code.

**Review by adamjstewart — COMMENTED** [review]
Knowing full well how different each SpaceNet version is, I won't try to ask you to combine the data.py to avoid code duplication. Hopefully someday we can have data modules for all versions.

File: tests/datasets/test_spacenet.py
Comment (nilsleh) [reply tests/datasets/test_spacenet.py:1]: Was a bit more involved, but I think it works now, except for mypy...

Comment (adamjstewart) [reply tests/datasets/test_spacenet.py:1]: Easy fix, it's an instance, not a class

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**